tastypie/resources.py

```diff
@@ -2054,6 +2065,35 @@ def filter_value_to_python(self, value, field_name, filters, filter_expr,
 
         return value
 
+    def get_query_terms(self, field_name):
+        """ Helper to determine supported filter operations for a field """
+
+        if field_name not in self.fields:
+            raise InvalidFilterError("The '%s' field is not a valid field" % field_name)
+
+        try:
+            django_field_name = self.fields[field_name].attribute
+            django_field = self._meta.object_class._meta.get_field(django_field_name)
+            if hasattr(django_field, 'field'):
+                django_field = django_field.field  # related field
+        except FieldDoesNotExist:
+            raise InvalidFilterError("The '%s' field is not a valid field name" % field_name)
+
+        return django_field.get_lookups().keys()
+
+    def resolve_filter_type(self, field_name, filter_bits, default_filter_type=None):
+        """ Helper to derive filter type from next segment in filter bits """
+
+        if not filter_bits:
+            # No filter type to resolve, use default
+            return default_filter_type
+        elif filter_bits[0] not in self.get_query_terms(field_name):
+            # Not valid, maybe related field, use default
+            return default_filter_type
```

Comment:
@dharvey-consbio there is an issue hidden here: it is incorrect to return the `default_filter_type` here for all the cases. 

With the examples of several levels of nested relations like `post__author__in` or `post__author__name__startswith`, the requests will fail to return the expected result because the default value of `'exact'` will be used here. Instead, we should recursively descend the chain of the `filter_bits`, similarly to how it is done in the `check_filtering` method above, and use the proper "filter type" for the last field in the chain. See the example of the fix implementation here: https://github.com/Partender/django-tastypie/pull/1/files/